homeassistant/components/ihc/auto_setup.py

```diff
@@ -105,36 +108,47 @@ def autosetup_ihc_products(hass: HomeAssistant, config, ihc_controller, controll
     groups = project.findall(".//group")
     for platform in IHC_PLATFORMS:
         platform_setup = auto_setup_conf[platform]
-        discovery_info = get_discovery_info(platform_setup, groups, controller_id)
+        discovery_info = get_discovery_info(
+            platform_setup, groups, controller_id, use_groups
+        )
         if discovery_info:
             discovery.load_platform(hass, platform, DOMAIN, discovery_info, config)
 
     return True
 
 
-def get_discovery_info(platform_setup, groups, controller_id):
+def get_discovery_info(platform_setup, groups, controller_id, use_groups: bool):
     """Get discovery info for specified IHC platform."""
     discovery_data = {}
     for group in groups:
         groupname = group.attrib["name"]
         for product_cfg in platform_setup:
             products = group.findall(product_cfg[CONF_XPATH])
             for product in products:
+                product_id = int(product.attrib["id"].strip("_"), 0)
                 nodes = product.findall(product_cfg[CONF_NODE])
                 for node in nodes:
                     if "setting" in node.attrib and node.attrib["setting"] == "yes":
                         continue
                     ihc_id = int(node.attrib["id"].strip("_"), 0)
                     name = f"{groupname}_{ihc_id}"
+                    model = product.get("product_identifier") or ""
+                    # make the model number look a bit nicer - strip leading _
+                    if model.startswith("_"):
+                        model = model[1::]
                     device = {
                         "ihc_id": ihc_id,
                         "ctrl_id": controller_id,
                         "product": {
+                            "id": product_id,
                             "name": product.get("name") or "",
                             "note": product.get("note") or "",
                             "position": product.get("position") or "",
+                            "model": model,
                         },
                         "product_cfg": product_cfg,
                     }
+                    if use_groups:
```

Comment:
I was reading the config flow PR to get a better understanding of this change, and it seems like the intent is to allow the user to disable suggested areas for devices. Do you have specific reasons for wanting to allow users to disable this?

My suggestion would be to keep this as simple as possible, limit configuration options unless absolutely needed, and always use the suggested areas, then let the user modify in the areas editor or existing devices editor since that already exists.

Unless you have reason to believe the groups are always pretty unreliable?

Reply:
My idea about the groups should be optional was because i did not want existing used to see any change, but since this is a "suggested" area it will not overwrite. Making it more simple is always good, so i will remove this option here and from the config flow also.